abi/BUILD

```diff
@@ -0,0 +1,134 @@
+# Standard constraint_setting and constraint_values to be used in platforms.
+licenses(["notice"])
+
+package(
+    default_visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "srcs",
+    srcs = glob(["**"]),
+)
+
+# To add a new constraint_value see https://github.com/bazelbuild/platforms.
+constraint_setting(name = "abi")
+
+constraint_value(
+    name = "atmega328",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "eabi",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "eabihf",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "elf",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "gnu",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "gnu_ilp32",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "gnueabi",
+    constraint_setting = ":abi",
+)
+
+constraint_value(
+    name = "gnuabi64",
```

Comment:
Is this a typo?

Reply:
No, this would match triples such as:
```
mips64-unknown-linux-gnuabi64
mips64el-unknown-linux-gnuabi64
```

Happy to remove this though and have more deeper conversation specifically about `gnu` variants.

Reply:
Got it. I've never looked at MIPS triples before :)